.github/workflows/terraform-plan-apply.yml

```diff
@@ -312,6 +309,27 @@ jobs:
           # mask this value so this sensitive data won't be logged.
           echo "::add-mask::$ui_storage_conn_string"
 
+      - name: Azure login
+        run: |
+          az login --service-principal -u ${ARM_CLIENT_ID} -p ${ARM_CLIENT_SECRET} --tenant ${ARM_TENANT_ID}
+          az account set --subscription ${ARM_SUBSCRIPTION_ID}
+      - name: Init apply permission
+        run: |
+          set -o errexit
+          set -o pipefail
+          set -o nounset
+          # set -o xtrace
+          # '-raw' is important so no other characters will be added
+          init_service_name=$(terraform output -raw init_service_name)
+
+          backend_object_id=$(az ad sp list --display-name 'mts-backend' --query "[0] .objectId")
+          app_role_id=$(az ad sp list --display-name 'mts-backend' --query "[0] .appRoles [?value=='internal.service'].id | [0]")
+          sp_object_id=$(az ad sp list --display-name $init_service_name --query "[0] .objectId")
```

Comment:
I see that Terraform exports the "principal_id" of the identity created. Isn't that the same as the one you're using here?